pkg/kn/commands/source/human_readable_flags.go

```diff
@@ -85,3 +101,148 @@ func printSourceTypesList(sourceTypesList *unstructured.UnstructuredList, option
 	}
 	return rows, nil
 }
+
+// printSource populates a single row of source list table
+func printSource(source unstructured.Unstructured, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	name := source.GetName()
+	sourceType := source.GetKind()
+	sourceTypeName := getSourceTypeName(source)
+	sink := findSink(source)
+	ready := isReady(source)
+
+	row := metav1beta1.TableRow{
+		Object: runtime.RawExtension{Object: &source},
+	}
+
+	if options.AllNamespaces {
+		row.Cells = append(row.Cells, source.GetNamespace())
+	}
+
+	row.Cells = append(row.Cells, name, sourceType, sourceTypeName, sink, ready)
+	return []metav1beta1.TableRow{row}, nil
+}
+
+// printSourceList populates the source list table rows
+func printSourceList(sourceList *unstructured.UnstructuredList, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	rows := make([]metav1beta1.TableRow, 0, len(sourceList.Items))
+
+	sort.SliceStable(sourceList.Items, func(i, j int) bool {
+		return sourceList.Items[i].GetName() < sourceList.Items[j].GetName()
+	})
+	for _, item := range sourceList.Items {
+		row, err := printSource(item, options)
+		if err != nil {
+			return nil, err
+		}
+
+		rows = append(rows, row...)
+	}
+	return rows, nil
+}
+
+func findSink(source unstructured.Unstructured) string {
+	sourceType := source.GetKind()
+	sourceJSON, err := source.MarshalJSON()
+	if err != nil {
+		return ""
+	}
+
+	switch sourceType {
+	case "ApiServerSource":
+		var apiSource eventinglegacy.ApiServerSource
+		err := json.Unmarshal(sourceJSON, &apiSource)
+		if err != nil {
+			return ""
+		}
+		return sinkToString(apiSource.Spec.Sink)
+	case "CronJobSource":
+		var cronSource eventinglegacy.CronJobSource
+		err := json.Unmarshal(sourceJSON, &cronSource)
+		if err != nil {
+			return ""
+		}
+		return sinkToString(cronSource.Spec.Sink)
+	case "SinkBinding":
+		var binding sourcesv1alpha1.SinkBinding
+		err := json.Unmarshal(sourceJSON, &binding)
+		if err != nil {
+			return ""
+		}
+		return flags.SinkToString(binding.Spec.Sink)
+	case "PingSource":
+		var pingSource sourcesv1alpha1.PingSource
+		err := json.Unmarshal(sourceJSON, &pingSource)
+		if err != nil {
+			return ""
+		}
+		return flags.SinkToString(*pingSource.Spec.Sink)
+	// TODO: Find out how to find sink in untyped sources
+	default:
+		return "<unknown>"
+	}
+}
+
+func isReady(source unstructured.Unstructured) string {
```

Comment:
Same here. Whether the object is ready or not should be already extracted before the printing method is called.